fili-core/src/test/groovy/com/yahoo/bard/webservice/web/endpoints/JobsServletSpec.groovy

```diff
@@ -155,6 +201,33 @@ class JobsServletSpec extends Specification {
         GroovyTestUtils.compareJson(result, expectedResponse, JsonSortStrategy.SORT_BOTH)
     }
 
+    def "jobs/ticket3p/results returns the number of results we requested through pagination parameters" () {
+        setup:
+        when: "We send a request to the jobs/ticket3p/results endpoint"
+        String result1 = makeRequest("/jobs/ticket3p/results", [asyncAfter : ["5"], perPage : [1], page: [1] ])
+
+        then: "We get only first row as we requested against total number of rows"
+        GroovyTestUtils.compareJson(result1, getExpectedFirstPage(), JsonSortStrategy.SORT_BOTH)
+
+        when: "We send a request to the jobs/ticket3p/results endpoint"
+        String result2 = makeRequest("/jobs/ticket3p/results", [asyncAfter : ["5"], perPage : [1], page: [3] ])
+
+        then: "We get last row as we requested against total number of rows"
+        GroovyTestUtils.compareJson(result2, getExpectedLastPage(), JsonSortStrategy.SORT_BOTH)
+
+        when: "We send a request to the jobs/ticket3p/results endpoint"
```

Comment:
Unless these comments are different in some way to indicate the steps happening, they are not providing any benefit. We should either remove these noisy comments, or we should update them to reflect what's going on in the `when` blocks